config/manager/manager.yaml

```diff
@@ -62,6 +62,13 @@ spec:
         - /manager
         args:
         - --leader-elect
+        env:
+        - name: WATCH_NAMESPACE
+          # This restricts the Operator to watch objects in the same
+          # namespace where it is deployed.
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
```

Comment:
I think we will want to have two versions of the operator manifest, or just have this snippet be mentioned in the docs for how to constrain it to its own namespace.

Reply:
Yeah, the `operator-sdk` docs are the only place that mention how to make the permissions dynamic - but it only covers the Golang part, not the RBAC manifests part. So we can have two, or just a `kustomize` command to inject it, or leave it commented out and just hard-code it when needed. Will tinker with it.

Reply:
Opened https://github.com/PrefectHQ/prefect-operator/issues/72 to track this.